src/prefs.js

```diff
@@ -235,39 +235,36 @@ export default class SensorsPreferences
         return row;
     }
 
-    _getSensorsLabels() {
-        let sensors_cmd = Utilities.detectSensors();
+    async _getSensorsLabels() {
+        const sensors_cmd = Utilities.detectSensors();
         if(sensors_cmd) {
-            let [result, sensors_output] = GLib.spawn_command_line_sync(sensors_cmd.join(' '));
-            if(result)
+            const [sensors_output, hasError] = await SubProc.runCommandAsync(sensors_cmd);
+            if(sensors_output)
             {
-                const decoder = new TextDecoder();
-                let output = decoder.decode(sensors_output);
```

Comment:
The communicate_utf8_async method returns suitable text so TextDecoder no longer required